src/sentry/integrations/github_enterprise/webhook.py

```diff
@@ -115,61 +112,67 @@ def get_secret(self, event, host):
             return None
 
     def handle(self, request: Request) -> Response:
-        body = bytes(request.body)
-        if not body:
-            logger.warning("github_enterprise.webhook.missing-body", extra=self.get_logging_data())
-            return HttpResponse(status=400)
-
-        try:
-            handler = self.get_handler(request.META["HTTP_X_GITHUB_EVENT"])
-        except KeyError:
-            logger.warning("github_enterprise.webhook.missing-event", extra=self.get_logging_data())
-            logger.exception("Missing Github event in webhook.")
-            return HttpResponse(status=400)
-
-        if not handler:
-            return HttpResponse(status=204)
-
-        try:
-            event = json.loads(body.decode("utf-8"))
-        except json.JSONDecodeError:
-            logger.warning(
-                "github_enterprise.webhook.invalid-json",
-                extra=self.get_logging_data(),
-                exc_info=True,
-            )
-            logger.exception("Invalid JSON.")
-            return HttpResponse(status=400)
-
-        try:
-            host = request.META["HTTP_X_GITHUB_ENTERPRISE_HOST"]
-        except KeyError:
-            logger.warning("github_enterprise.webhook.missing-enterprise-host")
-            logger.exception("Missing enterprise host.")
-            return HttpResponse(status=400)
-
-        secret = self.get_secret(event, host)
-        if not secret:
-            logger.warning("github_enterprise.webhook.missing-integration", extra={"host": host})
-            return HttpResponse(status=400)
-
-        try:
-            # Attempt to validate the signature. Older versions of
-            # GitHub Enterprise do not send the signature so this is an optional step.
-            method, signature = request.META["HTTP_X_HUB_SIGNATURE"].split("=", 1)
-            if not self.is_valid_signature(method, body, secret, signature):
+        with configure_scope() as scope:
+            meta = request.meta
+            try:
+                # XXX: There's lost of customers that are giving us an IP rather than a host name
+                # Use HTTP_X_REAL_IP in a follow up PR
+                host = meta["HTTP_X_GITHUB_ENTERPRISE_HOST"]
+            except KeyError:
+                logger.warning("github_enterprise.webhook.missing-enterprise-host")
+                logger.exception("Missing enterprise host.")
+                return HttpResponse(status=400)
```

Comment:
I move this block to the top of the function since we want to tag the host ASAP since we can use it for debugging.